pkg/builddef/set.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 The OSS Rebuild Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builddef
+
+import (
+	"context"
+
+	billy "github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/memfs"
+	git "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/google/oss-rebuild/pkg/rebuild/rebuild"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// BuildDefinitionSet represents a collection of build definitions.
+type BuildDefinitionSet interface {
+	Get(ctx context.Context, target rebuild.Target) (rebuild.Strategy, error)
```

Comment:
Is rebuild.Strategy sufficient for all our future build definition needs? Maybe it would make sense to have a BuildDefinition type that includes a strategy, but leaves the door open for adding other build definition info (information about who originally authored this definition and if it was manual or inferred, we also talked about the build definition having a "most recent status" field).

There are probably other fields that a build def might need that I can't think of right now, but making container with a strategy inside would make it much easier to add those things in the future, without costing much in the short term.

Reply:
We've discussed this previously. I think it's fine for now (notably, we don't yet have any use-cases to add metadata here) and this interface could easily be updated to return the new construct if/when we migrate.